src/server/scripts/Commands/PlayerCommand.cpp

```diff
@@ -44,7 +44,7 @@ bool PlayerCommand::Learn(ChatHandler* handler, Player* targetPlayer, uint32 spe
 
     if (!allRanks && targetPlayer->HasSpell(spell))
     {
-        if (targetPlayer == handler->GetSession()->GetPlayer())
+        if (handler->GetSession() && targetPlayer == handler->GetSession()->GetPlayer())
```

Comment:
Just for my understanding... Why do we check GetSession twice?

Reply:
The first check is out of scope, and it will not return the function all the time. So a new check is necessary. We could move the block up but IMO it would make it a bit harder to read.

Reply:
as long as it works im fine with it